edgecore will panic if we run a pod with device plugin resource

<!-- Please use this template while reporting a bug and provide as much info as possible. Thanks!-->


**What happened**:edgecore will panic if we run a pod with device plugin resource, panic info:
```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x24e1892]

goroutine 111 [running]:
github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager.(*ManagerImpl).updateAllocatedDevices(0xc000401ae0, 0xc000912050, 0x2, 0x2)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager/manager.go:569 +0x42
github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager.(*ManagerImpl).allocateContainerResources(0xc000401ae0, 0xc000256000, 0xc000cb9348, 0xc000cb9248, 0xc0000fbcb0, 0x0)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager/manager.go:669 +0xecb
github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager.(*ManagerImpl).allocatePodResources(0xc000401ae0, 0xc000256000, 0x2b859a0, 0xc00094fbc0)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager/manager.go:331 +0x269
github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager.(*ManagerImpl).Allocate(0xc000401ae0, 0xc00052ef70, 0xc000777800, 0x0, 0x0)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/devicemanager/manager.go:343 +0x55
github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm.(*containerManagerImpl).UpdatePluginResources(0xc0004eb680, 0xc00052ef70, 0xc000777800, 0xc00052ef70, 0x1)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cm/container_manager_linux.go:646 +0x4e
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).addPod(0xc00032a840, 0x2e88200, 0xc000256000)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:1029 +0x1ce
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).handlePodListFromMetaManager(0xc00032a840, 0xc000bcc000, 0x59d7, 0x6000, 0x1, 0x1)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:1002 +0x17b
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).syncPod(0xc00032a840)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:918 +0x7f9
github.com/kubeedge/kubeedge/edge/pkg/edged.(*edged).Start(0xc00032a840, 0xc000776d60)
        /home/zbq/code/src/github.com/kubeedge/kubeedge/edge/pkg/edged/edged.go:318 +0xd61
created by github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core.StartModules
        /home/zbq/code/src/github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core/core.go:22 +0x166
```


**What you expected to happen**: pod with device plugin resource can run by edgecore

**How to reproduce it (as minimally and precisely as possible)**:
1.update edge.yaml device-plugin-enabled to true

```bash
edged:
    register-node-namespace: default
    hostname-override: fb4ebb70-2783-42b8-b3ef-63e2fd6d242e
    interface-name: eth0
    edged-memory-capacity-bytes: 7852396000
    node-status-update-frequency: 10 # second
    device-plugin-enabled: true
    gpu-plugin-enabled: true
```

2. run edgecore

3.run device plugin

4.create a pod with device plugin resource

**Anything else we need to know?**:

**Environment**:
- KubeEdge version:v1.15.0-kubeedge-v1.0.0
- Hardware configuration: x86 vm
- OS (e.g. from /etc/os-release): Ubuntu 16.04.5
- Kernel (e.g. `uname -a`):4.4.0-141
- Others:

--- a/edge/pkg/edged/edged.go
+++ b/edge/pkg/edged/edged.go
@@ -609,7 +609,7 @@ func newEdged() (*edged, error) {
 
 func (e *edged) initializeModules() error {
 	node, _ := e.initialNode()
-	if err := e.containerManager.Start(node, e.GetActivePods, nil, e.statusManager, e.runtimeService); err != nil {
+	if err := e.containerManager.Start(node, e.GetActivePods, edgedutil.NewSourcesReady(), e.statusManager, e.runtimeService); err != nil {
 		log.LOGGER.Errorf("Failed to start device plugin manager %v", err)
 		return err
 	}

